src/libraries/Pool.sol

```diff
@@ -338,6 +346,8 @@ library Pool {
         });
 
         StepComputations memory step;
+        swapFee =
+            cache.protocolFee == 0 ? slot0Start.lpFee : uint24(cache.protocolFee).calculateSwapFee(slot0Start.lpFee);
```

Comment:
do we have any more stack room in this function? 
would love to see `slot0Start.lpFee` and `cache.protocolFee` cached. But I realise i'm probably asking for too much 💀

Reply:
thanks @shuhuiluo did the refactor to decrease bytecode size

Reply:
I think caching `params.sqrtPriceLimitX96` saves more.